enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,891 @@
+| Machine API | `ClusterAPI`| Create | Yes | - | Sync controller syncs Cluster API spec to Machine API Machine |
+| Machine API | `ClusterAPI`| Create | No | - | Sync controller creates Cluster API equivalent and starts synchronisation from Cluster API to Machine API for future updates |
+| Machine API | `''` | Create | Yes | - | Annotation is set to `ClusterAPI` and follows above description |
+| Machine API | `''` | Create | No | - | Annotation is set to `MachineAPI` and follows above description |
+| Machine API | - | Delete | Yes | Yes | Sync controller deletes Cluster API equivalent Machine |
+| Machine API | - | Delete | Yes | No | Cluster API equivalent Machine remains in cluster |
+| Cluster API | - | Create | N/A | - | Sync controller takes no action. Cluster API Machine acts independently of Machine API |
+| Cluster API | `MachineAPI`  | Create | Yes | - | Sync controller syncs Machine API spec to Cluster API Machine |
+| Cluster API | `ClusterAPI`  | Create | Yes | - | Sync controller syncs Cluster API spec to Machine API Machine |
+| Cluster API | - | Delete | Yes | Yes | Sync controller deletes Machine API equivalent Machine |
+| Cluster API | - | Delete | Yes | No | Machine API synchronisation controller will recreate Machine after it is removed[^9] |
+
+[^9]: Any Cluster API Machine that is deleted that does not have an owner, but is reflecting a Machine in Machine API,
+is likely to be a control plane Machine without a ControlPlaneMachineSet.
+In this case, to prevent accidental deletion, we do not propagate the delete event.
+This should not interfere with the semantics of any higher level abstraction or resource such as MachineHealthCheck.
+
+#### Pausing of Cluster API controllers
+
+When a resource is set to be authoritative in Machine API, the Cluster API controllers should be paused.
+
+Cluster API provides either an annotation (`cluster.x-k8s.io/paused`) or field based (`.spec.paused`) approach to pausing controllers.
+
+Controllers in Cluster API respect the pause mechanism to prevent them from taking any action on the resource.
+This matches the requirements for the non-authoritative resources described in this document.
+
+The migration controller will leverage the pause mechanism to ensure that the Cluster API controllers do not take any action on the resource when it is non-authoritative.
+Admission control will ensure that the value cannot be changed by users when the resource is non-authoritative.
+
+#### Filtering reconciles in Machine API controllers
+
+Each controller will, at the start of the reconcile logic, check for the presence of the `status.authoritativeAPI` and will either skip or continue with the reconciliation based on the presence of this field.
+
+The controller will set the `Paused` condition to `True` when the resource is non-authoritative and will not perform any further reconciliation.
+
+#### The Cluster Autoscaler in mixed MachineSet clusters
```

Comment:
@elmiko Have you had a chance to look over this for consistency? I know you've looked at ensuring we respect upstream autoscaler values, so hopefully that will help us transition smoothly.

Reply:
We discussed this some in a standup, and the autoscaler's behavior will be expanded on further in a separate enhancement, so I'd say this doesn't need to be blocking.

Reply:
we talked about the autoscaler usage in our standup yesterday. in general i think we are ok to go forward with this design but there are a few details to keep in mind:

* today, the autoscaler is configured to watch a specific group and version for the scalable types.
* given that we will allow per-machineset upgrading into capi, we will most likely need to carry a patch on the autoscaler that will allow us to differentiate between the various resource types we are seeing.
* there will need to be some design documented about how the autoscaler logic will work in terms of selecting the proper resource to scale. this can be handled in a followup enhancement specifically about the autoscaler.